build_tools/github_actions/build_dist.py

```diff
@@ -179,17 +179,46 @@ def build_py_runtime_pkg():
       f"-DIREE_BUILD_PYTHON_BINDINGS=ON",
       f"-DIREE_BUILD_SAMPLES=OFF",
       f"-DIREE_BUILD_TESTS=OFF",
-  ])
+  ],
+                 check=True)
 
   print("*** Building ***")
-  subprocess.check_call([
+  subprocess.run([
       sys.executable,
       CMAKE_CI_SCRIPT,
       "--build",
       BUILD_DIR,
       "--target",
       "install-IreePythonPackage-rt-stripped",
-  ])
+  ],
+                 check=True)
+
+
+def bazel_build_tf_binary(target):
+  subprocess.run([
+      "bazel",
+      "build",
+      "--config=release",
+      target,
+  ],
+                 cwd=TF_INTEGRATIONS_DIR,
+                 check=True,
+                 universal_newlines=True)
+
+  process = subprocess.run(["bazel", "info", "bazel-bin"],
+                           cwd=TF_INTEGRATIONS_DIR,
+                           check=True,
+                           capture_output=True,
+                           universal_newlines=True)
+  if len(process.stdout.splitlines()) != 1:
+    raise RuntimeError(
+        f"Unexpected output from `bazel info bazel-bin`:\n{process.stdout}")
+  bazel_bin_dir = process.stdout.strip()
+  if not os.path.isdir(bazel_bin_dir):
+    raise RuntimeError(
+        f"`bazel info bazel-bin` '{bazel_bin_dir}' is not a directory")
+
+  return bazel_bin_dir
```

Comment:
Can you document this (either through variable naming or comments) a bit more?
* I'm not sure why `bazel_build_tf_binary(target)` is returning a directory path
* `process` could be renamed to `info_process` or something similarly descriptive, considering there is another `subprocess.run()` call in this function that does something different

You could also split this into two functions - one for `bazel build` and one for `bazel info bazel-bin`

Reply:
Yeah you're right this is not great. I needed to call it always in the same place and from the same working directory so I merged them into one function. I'll rework it

Reply:
Ok I made the function just return the path to the binary it just built, which seems like a much more reasonable thing for it to do. Of course, this totally reasonable thing is not well supported in Bazel, but :roll_eyes: